Add new functions

MinASL19 opened the pull request:
1. asType
2. constructUrl
3. getAbsoluteUrl
4. getFirstStringOrString
5. getNormalizedQuery
6. getObjectValueByDotSeparatedKey
7. logError

MinASL19 committed:
add new functions

MinASL19 committed:
Merge branch 'develop' into feature/1-add-new-functions

MinASL19 committed:
update dependencies and add more functions

MinASL19 committed:
Merge branch 'feature/1-add-new-functions' of github.com:ASL-19/js-utils into feature/1-add-new-functions

MinASL19 committed:
merge

MinASL19 committed:
add eslint config

MinASL19 committed:
fix eslint value

MinASL19 committed:
update settings

GrantASL19 committed:
Misc. changes:

- Added Prettier config and updated formatting of existing files

- Added some missing dev dependencies

- Update some function documentation: moved some @remarks up to reduce the length of docs written to `docs/js-utils.md`; added @example in some places; update code example formatting so it gets proper TS code formatting; etc. (most of this is new stuff based on my own research while reviewing)

- Wrapped some comments/docstrings at 80 columns using VS Code Rewrap extension opt-Q shortcut (https://marketplace.visualstudio.com/items?itemName=stkb.rewrap)

- Replaced Yarn configs with NPM (forgot this package still had Yarn!)

- Updated README based on new @asl-19/js-dom-utils README

GrantASL19 reviewed (commented):
@MinASL19 Thanks, didn’t realize how many functions we’d be able to pull into this!

Just added some updates in 9a926979e8183f951d635f48543183e2ac4f233d. Can you take a look a these, and test in the internal project MR I just assigned you to?

GrantASL19 committed:
Update getObjectValueByDotSeparatedKey; rename logError to serverLog and adjust formatting

In getObjectValueByDotSeparatedKey I improved the type safety. You now
need to pass a generic leaf node
(https://en.wikipedia.org/wiki/Tree_(data_structure)) type parameter to
it. I also clarified that this is meant to be used only on the server —
it will now log an error if called from the browser.

In internal projects we’ll probably wrap this in most cases (see
getStringByDotSeparatedKey in the project we’re testing this in).

GrantASL19 committed:
Update getAbsoluteUrl console.warn prefixes

GrantASL19 committed:
Run build-lint-test in GitHub action

GrantASL19 committed:
Move ts-pattern to dependencies
